File: StarMines/Data/Python/blue.py
import math

def behave():
    b = Behavior()

    #check calls

    call = b.getCall()
    while call:
        if call.contents.label == b'infect':
            infect(b)
            break

        call = b.getCall()

    #movement

    counter = b.getRegisterInt(0)
    counter += 1
    if counter == 15:
        b.setRegisterInt(0,0)
        move(b)
    else:
        b.setRegisterInt(0,counter)

    #bounds
    pos = b.position()
    x = pos.contents.x
    y = pos.contents.y

    if x > 770:
        x = 770
    if x < 0:
        x = 0
    if y > 570:
        y = 570
    if y < 0:
        y = 0

    b.setPosition(x,y)

def move(b):
    
    pos = b.position()
    x = pos.contents.x
    y = pos.contents.y

    b.setTargetByTag(b'Red', b'near')
    
    tpos = b.targetPosition()

    if not tpos:
        return

    u = tpos.contents.x
    v = tpos.contents.y

    if math.fabs(u-x) > 50 or math.fabs(v-y) > 50:
        b.setDirection(0,0)
        return

    b.setSpeed(3)
    b.setDirection(u-x, v-y)


def infect(b):
    pos = b.position()
    x = pos.contents.x
    y = pos.contents.y

    b.spawn(b'Red', x, y, None, 0)
    b.despawn(None, 0)

    b.playSound(b'infect',40,False,False)
    b.deactivate()

behave()

File: StarMines/TestData/Pandemic/Lua/Blue.lua
function Blue_bounds(bc)
	local pos = bc:position()
	local x = pos.x
	local y = pos.y
	if x > 770 then x = 770 end
	if x < 0 then x = 0 end
	if y > 570 then y = 570 end
	if y < 0 then y = 0 end
	bc:setPosition(x,y)
end

function Blue_escape(bc)
	bc:setRegInt(0, 30)
	bc:setTargetByTag("Red", "near")
	local pos = bc:position()
	local tpos = bc:targetPosition()
	local x,y,tx,ty = pos.x, pos.y, tpos.x, tpos.y
	if math.abs(x - tx) > 100 or math.abs(y - ty) > 100 then bc:setSpeed(0) return end
	local dx,dy = 0,0
	if x < tx then dx = -1 end
	if x > tx then dx = 1 end
	if y < tx then dy = -1 end
	if y > ty then dy = 1 end
	bc:setDir(dx,dy)
	bc:setSpeed(3)
end

function Blue_move(bc)
	local counter = bc:getRegInt(0)
	if counter == 0 then Blue_escape(bc) end
	bc:setRegInt(0, bc:getRegInt(0) - 1)
	Blue_bounds(bc)
end

function Blue_calls(bc)
	local call = bc:getCall()
	while call.label ~= "nil" do
		if call.label == "collision" and call.tag == "Red" then
			local pos = bc:position()
			bc:spawn("Red", pos.x, pos.y, "")
			bc:despawn("")
			bc:playSound("Infect", 40, false, false)
			return
		end
		call = bc:getCall()
	end
end

function Blue(bc)
	Blue_calls(bc)
	Blue_move(bc)
end